CMakeLists.txt

```diff
@@ -329,8 +343,12 @@ endif()
 
 # In release mode, we turn on even more optimizations if we do not have to provide a portable binary.
 if (NOT STORM_PORTABLE)
-	set (CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=native")
-endif ()
+	if(NOT APPLE_SILICON OR (STORM_COMPILER_CLANG AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 15.0))
+		set (CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=native")
+	elseif(STORM_COMPILER_APPLECLANG AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13.0)
+		set (CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -mcpu=apple-m1")
```

Comment:
Does this break if people have other ARM processors, like Apple M2?

Reply:
I don't think it does for M2. Some posts from before the `-mcpu` definition for M1 existed already suggested using the A-series chip (iPhone/iPad) defintion, so using M1 here shouldn't break anything. Unfortunately, I don't have a machine to test it on :)
Other ARM processors might have a problem, depending on whether `-march=native` is supported for the platform. I would leave it like this for now, if a problem occurs, we know where to look for it.